src/Microsoft.PowerShell.Commands.Utility/commands/utility/Get-Error.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// Class for Get-Error implementation.
+    /// </summary>
+    [Experimental("Microsoft.PowerShell.Utility.PSGetError", ExperimentAction.Show)]
+    [Cmdlet(VerbsCommon.Get, "Error", HelpUri = "https://docs.microsoft.com/powershell/module/microsoft.powershell.utility/get-error?view=powershell-7&WT.mc_id=ps-gethelp", DefaultParameterSetName = NewestParameterSetName)]
+    public sealed class GetErrorCommand : PSCmdlet
+    {
+        internal const string ErrorParameterSetName = "Error";
+        internal const string NewestParameterSetName = "Newest";
+        internal const string AliasNewest = "Last";
+
+        /// <summary>
+        /// Gets or sets the error object to resolve.
+        /// </summary>
+        [Parameter(Position = 0, ValueFromPipeline = true, ParameterSetName = ErrorParameterSetName)]
+        public PSObject InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the number of error objects to resolve starting with newest first.
+        /// </summary>
+        [Parameter(ParameterSetName = NewestParameterSetName)]
+        [Alias(AliasNewest)]
+        [ValidateRange(1, int.MaxValue)]
+        public int Newest { get; set; } = 1;
+
+        /// <summary>
+        /// Process the error object.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            var errorRecords = new List<object>();
+            var index = 0;
+
+            if (InputObject != null)
+            {
+                if (InputObject.BaseObject is Exception || InputObject.BaseObject is ErrorRecord)
+                {
+                    errorRecords.Add(InputObject);
+                }
+            }
+            else
+            {
+                var errorVariable = SessionState.PSVariable.Get("error");
+                var count = Newest;
+                ArrayList errors = (ArrayList)errorVariable.Value;
+                if (count > errors.Count)
+                {
+                    count = errors.Count;
+                }
+
+                while (count > 0)
+                {
+                    errorRecords.Add(errors[index]);
+                    index++;
+                    count--;
+                }
+            }
+
+            index = 0;
+            bool addErrorIdentifier = errorRecords.Count > 1 ? true : false;
+
+            foreach (object errorRecord in errorRecords)
+            {
+                PSObject obj = PSObject.AsPSObject(errorRecord);
+                obj.TypeNames.Insert(0, "System.Management.Automation.ErrorRecord#GetError");
```

Comment:
Is using the command name that adds the data to the object as the ETS identifier the best practice? I've seen this done elsewhere, but I don't think it's the best approach. What if I were to write a module for error management, and I wanted to add the same data to one of my functions? Should I use a different ETS name when I write it to contain the same information? That wouldn't be the right thing to do. IMHO the ETS name should identify what's different about the object from the original version. That's somewhat ambiguous and subject to interpretation, but it's better than using a command name, since once you have the data, it shouldn't matter where it came from -- you should just be able to do things with it: use the ETS name as a parameter in a function that accepts a specific ETS type, create custom formatters for it, use it in tooling, etc. Personally I'd prefer you use a name like `System.Management.Automation.ErrorRecord#PSExtended`, and reserve '#PSExtended' for extensions from PowerShell itself so that you don't need to worry about stepping on the toes of other users, while still keeping the ETS identifier generic enough that it works more broadly than a single command.

Reply:
Discussed this with @JamesWTruher and we agree with your feedback and decided on "PSExtendedError" as the typename